src/main/java/net/sf/jabref/gui/importer/actions/OpenDatabaseAction.java

```diff
@@ -171,111 +173,71 @@ private void openTheFile(File file, boolean raisePanel) {
-                } catch (IOException ex) {
-                    LOGGER.error("Error loading database " + fileToLoad, ex);
-                    result = ParserResult.getNullResult();
-                }
-                if (result.isNullResult()) {
-                    JOptionPane.showMessageDialog(null, Localization.lang("Error opening file") + " '" + fileName + "'",
-                            Localization.lang("Error"), JOptionPane.ERROR_MESSAGE);
+            }
 
-                    String message = "<html>" + errorMessage + "<p>"
-                            + (tryingAutosave ? Localization.lang(
-                                    "Error opening autosave of '%0'. Trying to load '%0' instead.",
-                                    file.getName()) : ""/*Globals.lang("Error opening file '%0'.", file.getName())*/)
-                            + "</html>";
-                    JOptionPane.showMessageDialog(null, message, Localization.lang("Error opening file"),
-                            JOptionPane.ERROR_MESSAGE);
-
-                    if (tryingAutosave) {
-                        tryingAutosave = false;
-                        fileToLoad = file;
-                    } else {
-                        done = true;
-                    }
-                    continue;
-                } else {
-                    done = true;
-                }
+            if (BackupManager.checkForBackupFile(fileToLoad.toPath())) {
+                BackupUIManager.showRestoreBackupDialog(frame, fileToLoad.toPath());
+            }
 
-                final BasePanel panel = addNewDatabase(result, file, raisePanel);
-                if (tryingAutosave) {
-                    panel.markNonUndoableBaseChanged();
-                }
+            ParserResult result;
+            try {
+                result = OpenDatabase.loadDatabase(fileToLoad, Globals.prefs.getImportFormatPreferences());
+            } catch (IOException ex) {
+                LOGGER.error("Error loading database " + fileToLoad, ex);
+                result = ParserResult.getNullResult();
+                JOptionPane.showMessageDialog(null, Localization.lang("Error opening file") + " '" + fileName + "'",
+                        Localization.lang("Error"), JOptionPane.ERROR_MESSAGE);
+            }
 
-                // After adding the database, go through our list and see if
-                // any post open actions need to be done. For instance, checking
-                // if we found new entry types that can be imported, or checking
-                // if the database contents should be modified due to new features
-                // in this version of JabRef:
-                final ParserResult finalReferenceToResult = result;
-                SwingUtilities.invokeLater(
-                        () -> OpenDatabaseAction.performPostOpenActions(panel, finalReferenceToResult, true));
+            if (Objects.nonNull(result.getDatabase().getDatabaseID())) {
```

Comment:
I already thought about this option, but this is unfortunately not possible as meta-data, preamble, etc. are also shared. This would end in disaster. This key/ID has to be stored in a local file but it also must not be in shared.